### PR TITLE
[JENKINS-54674] Add Java 11 into PCT docker image

### PR DESCRIPTION
**Pull request by alecharp**
- [x] Add openjdk binary to the docker image
- [x] Add JAXB libraries to docker image
- [x] Add activator to use alternate JVM to run PCT tests

@jenkinsci/java11-support @raul-arabaolaza

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -43,11 +43,26 @@ ENV INSTALL_BUNDLED_SNAPSHOTS=true
 
 RUN apt-get -y update && apt-get install -y groovy && rm -rf /var/lib/apt/lists/*
 
+RUN curl -L --show-error https://download.java.net/java/GA/jdk11/13/GPL/openjdk-11.0.1_linux-x64_bin.tar.gz --output openjdk.tar.gz && \
```
Comment (oleg-nenashev) [inline Dockerfile]: Would be nice to put 11.0.1 and 13 to ARG definitions so that we can update easily

Comment (alecharp) [reply Dockerfile]: For the moment, I don't know how the next update release will be named and how the URL would be. 

Also, we would have to change the `sha256`  to validate the binary. So when we will change the binary url, we would also have to change the checksum.

File: Dockerfile [hunk 2]
```diff
@@ -43,11 +43,26 @@ ENV INSTALL_BUNDLED_SNAPSHOTS=true
...
 
+ARG JAXB_API_VERSION="2.3.0"
+ARG JAXB_VERSION="2.3.0.1"
+ARG JAF_VERSION="1.2.0"
+RUN mkdir -p /pct/jdk11-libs && \
+    curl -L http://repo.jenkins-ci.org/public/javax/xml/bind/jaxb-api/${JAXB_API_VERSION}/jaxb-api-${JAXB_API_VERSION}.jar -o /pct/jdk11-libs/jaxb-api.jar && \
+    curl -L http://repo.jenkins-ci.org/public/com/sun/xml/bind/jaxb-core/${JAXB_VERSION}/jaxb-core-${JAXB_VERSION}.jar -o /pct/jdk11-libs/jaxb-core.jar && \
+    curl -L http://repo.jenkins-ci.org/public/com/sun/xml/bind/jaxb-impl/${JAXB_VERSION}/jaxb-impl-${JAXB_VERSION}.jar -o /pct/jdk11-libs/jaxb-impl.jar && \
+    curl -L http://repo.jenkins-ci.org/public/com/sun/activation/${JAF_VERSION}/javax.activation-${JAF_VERSION}.jar -o /pct/jdk11-libs/javax.activation.jar
+
```
Comment (batmat) [inline Dockerfile]: Pondering how we could centralize this somehow. Also nice to have, we could checksum these files to detect when/if a download went wrong (and hence avoid failing later in a probably less understandable way)

Comment (alecharp) [reply Dockerfile]: I can work on that.

Comment (alecharp) [reply Dockerfile]: done.

File: Dockerfile (hunk 1 above)
Comment (alecharp) [reply Dockerfile]: @oleg-nenashev I can still make the modification if you want. Do you think it's preferable?

**Comment by oleg-nenashev** [comment]
Some documentation would be nice tho.
IMHO it makes sense to just add it to https://github.com/jenkinsci/plugin-compat-tester#running-pct-with-custom-java-versions, but it may be a follow-up

File: src/main/docker/run-pct.sh [hunk 3]
```diff
@@ -123,9 +123,9 @@ mkdir -p "${PCT_OUTPUT_DIR}"
 ###
 # Determine if we test the plugin against another JDK
 ###
-if [[ "${JDK11}" ]] ; then
-  TEST_JDK_HOME="/usr/lib/jvm/java-11-opendjdk-amd64"
-  TEST_JAVA_ARGS="-p /pct/jdk11-libs/jaxb-api.jar:/pct/jdk11-libs/javax.activation.jar --add-modules java.xml.bind,java.activation -cp /pct/jdk11-libs/jaxb-impl.jar:/pct/jdk11-libs/jaxb-core.jar"
+TEST_JDK_HOME=${:-"/usr/lib/jvm/java-${JDK_VERSION:-8}-opendjdk-amd64"}
+if [[ "${JDK_VERSION}" = "11" ]] ; then
```
Comment (batmat) [inline src/main/docker/run-pct.sh]: Add a log maybe?

Comment (alecharp) [reply src/main/docker/run-pct.sh]: Noted.

Comment (alecharp) [reply src/main/docker/run-pct.sh]: Done.

**Pull request merged**